src/transaction.ts

```diff
@@ -2530,6 +2541,110 @@ function buildDeleteMutation(
   return mutation as spannerClient.spanner.v1.Mutation;
 }
 
+/**
+ * Mutation represent a set of changes to be applied atomically to a cloud spanner
+ * database with a {@link Transaction}.
+ * Mutations are used to insert, update, upsert(insert or update), replace, or
+ * delete rows within tables.
+ *
+ * Mutations are added to a {@link Transaction} and are not executed until the
+ * transaction is committed via {@link Transaction#commit}.
+ *
+ * If the transaction is rolled back or encounters an error, the mutations are
+ * discarded.
+ *
+ * @example
+ * ```
+ * const {Spanner, Mutation} = require('@google-cloud/spanner');
+ * const spanner = new Spanner();
+ *
+ * const instance = spanner.instance('my-instance');
+ * const database = instance.database('my-database');
+ *
+ * const mutation = new Mutation();
+ * mutation.insert('Singers', {SingerId: '123', FirstName: 'David'});
+ * mutation.update('Singers', {SingerId: '123', FirstName: 'Marc'});
+ *
+ * try {
+ *  database.writeAtLeastOnce(mutation, (err, res) => {
+ *    console.log("RESPONSE: ", res);
+ *  });
+ * } catch(err) {
+ *  console.log("ERROR: ", err);
+ * }
+ * ```
+ */
+export class Mutation {
```

Comment:
I find the name of this class a bit confusing. This does not represent a single `Mutation`, but a (mutable) set of mutations. Maybe rename to `MutationSet`?

Reply:
Updated